eval/adjust.rkt

```diff
@@ -89,10 +87,9 @@
 
     ; Final precision assignment based on the upper bound
     (define final-precision
-      (max (+ max-prec (vector-ref vstart-precs (- n varc))) (*base-tuning-precision*)))
-    (vector-set! vprecs-max (- n varc) (min final-precision (*rival-max-precision*)))
-    (when (>= final-precision (*rival-max-precision*))
-      (set! early-stopping-upper-bound #t))
+      (min (max (+ max-prec (vector-ref vstart-precs (- n varc))) (*rival-min-precision*))
+           (*rival-max-precision*)))
```

Comment:
It is here, it should be `(*rival-min-precision*)` instead of `(*base-tuning-precision*)`. They mean different things